vehicle/mkvi/software/throttle/throttle.c

```diff
@@ -0,0 +1,318 @@
+    if (*pos_1 > MAX_THROTTLE_POS) {
+        *pos_1 = MAX_THROTTLE_POS;
+        throttle.throttle_status = THROTTLE_1_OUT_OF_RANGE;
+        throttle_debug.throttle_1_out_of_range = true;
+        implausibility = true;
+    } else if (*pos_1 < MIN_THROTTLE_POS) {
+        *pos_1 = MIN_THROTTLE_POS;
+        throttle.throttle_status = THROTTLE_1_OUT_OF_RANGE;
+        throttle_debug.throttle_1_out_of_range = true;
+        implausibility = true;
+    } else {
+        throttle_debug.throttle_1_out_of_range = false;
+    }
+
+    // Check 2nd pot
+    if (*pos_2 > MAX_THROTTLE_POS) {
+        *pos_2 = MAX_THROTTLE_POS;
+        throttle.throttle_status = THROTTLE_2_OUT_OF_RANGE;
+        throttle_debug.throttle_2_out_of_range = true;
+        implausibility = true;
+    } else if (*pos_2 < MIN_THROTTLE_POS) {
+        *pos_2 = MIN_THROTTLE_POS;
+        throttle.throttle_status = THROTTLE_2_OUT_OF_RANGE;
+        throttle_debug.throttle_2_out_of_range = true;
+        implausibility = true;
+    } else {
+        // Will be reset if Ready to Drive is not on
+        throttle.throttle_status = THROTTLE_RUN;
+        throttle_debug.throttle_2_out_of_range = false;
+    }
+    return implausibility;
+}
+
+/*
+    Check for deviation between 2 throttle sensors, >10% deviation is an
+    implausibility
+    Args:
+        pos_min: int16_t bigger sensor value of the 2 throttle pots
+        pos_max: int16_t smaller sensor value of the 2 throttle pots
+    Returns:
+        boolean if there's deviation or not
+*/
+static bool check_deviation(int16_t pos_max, int16_t pos_min) {
+    if (pos_max - pos_min > APPS_IMPLAUSIBILITY_DEVIATION_THRESHOLD) {
+        throttle.throttle_status = THROTTLE_POSITION_IMPLAUSIBILITY;
+        throttle_debug.throttle_deviation = true;
+        return true;
+    } else {
+        throttle.throttle_status = THROTTLE_RUN;
+        throttle_debug.throttle_deviation = false;
+        return false;
+    }
+}
+
+/*
+    Check if brakes are on while pedal travel >= 25%, if so return
+    implausibility If pedal travel becomes <5%, clear implausibility.
+    Args:
+        pos_min: int16_t smaller sensor value of the 2 pots
+    Returns:
+        true if implausibility, false if not
+*/
+static bool check_brake(int16_t pos_min) {
+    static bool brake_implausibility_occurred = false;
+
+    if (throttle_state.brake_pressed) {
+        // if brakes are pressed
+        if (pos_min >= APPS_BRAKE_IMPLAUSIBILITY_THRESHOLD) {
+            // brake is pressed, pedal travel >= 25%
+            brake_implausibility_occurred = true;
+            throttle.throttle_status = THROTTLE_BRAKE_PRESSED;
+            throttle_debug.throttle_brake_implaus = true;
+            return true;
+        } else {
+            // brake is pressed, pedal travel < 25%
+            // no implausibility
+            throttle_debug.throttle_brake_implaus = false;
+            return false;
+        }
+
+        if (brake_implausibility_occurred) {
```

Comment:
Doesn't this if statement never get executed since the previous block is guaranteed to return something?